### PR TITLE
Halsema/multiple perf tests

### LINKED ISSUE #4134: Support reporting multiple performance testing types in PRs.
Current reporting only surfaces the baseline render time tests.

Combobox only tests one version of the combobox, so we can't see how it compares on a regular basis within the PRs. You have to run it locally. How can we expand this to include multiple types of usages in the tests?

Follow-up work: https://github.com/orgs/adobe/projects/50/views/2?pane=issue&itemId=53607460

Paths: 
1. Expand PR Tachometer comments to include comparisons against multiple types of tests (previous against current), akin to what you see when running the command in your terminal 
2. Test specific interactions

### PR DESCRIPTION
**Pull request by najikahalsema**
## Description

To-do: 
- [x] Confirm that a for loop is... an okay way to do this lol
- [x] Add the appropriate commit message type lol
- [x] add `--since origin/main` back into command line that I removed to test this PR

## Related issue(s)

- resolves #4134 

## Motivation and context

Preparation for DSRR 

## How has this been tested?

-  Via github actions 
- 
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: tasks/test-changes.js [hunk 1]
```diff
@@ -38,7 +38,8 @@ const getChangedPackages = () => {
                 // There are no benchmarks available in this directory.
                 item.location.search('projects') === -1 &&
                 // The icons-* tests are particular and long, exclude in CI.
-                !name.startsWith('icons-')
+                !name.startsWith('icons-') &&
+                !name.startsWith('color-field')
```
Comment (Westbrook) [inline tasks/test-changes.js:42]: Doesn't let me comment on line 51, but you could return `['overlay']` and hopefully only need to run a single test each time to see what the output would be.

File: tasks/build-tachometer-comment.js [hunk 2]
```diff
@@ -102,28 +102,75 @@ function formatDifference({ absolute, percentChange: relative }) {
...
+
+    let branch = result[1];
+    let branchDifferences = formatDifference(branch.differences[0]);
+    let branchDifferencesString = `${branchDifferences.label} <br> ${branchDifferences.relative} <br> ${branchDifferences.absolute}`;
+
     const packageName = `${result[0].name.split(':')[0]}`;
-    const table = `<a id="${packageName}"></a>
+    const table = [];
 
-## ${packageName} [_permalink_](#user-content-${packageName})
```
Comment (Westbrook) [inline tasks/build-tachometer-comment.js:114]: Be very conscious of the left hand spacing as it has a meaning in Markdown.

**Comment by najikahalsema** [comment]
Preserving the Tachometer comment that I used to test the overlay's different test types, since they will likeyl get overwritten once I add back `--since origin/main`.

------

## Tachometer results


<details>
    <summary><strong>Chrome</strong></summary>

<a id="overlay"></a>

## overlay [_permalink_](#user-content-overlay)


    
### basic-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 672 kB | 468.53ms - 474.13ms | - | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -5.18ms - +4.87ms |
| branch | 663 kB | 467.32ms - 475.66ms | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -4.87ms - +5.18ms | - |


    
### directive-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 788 kB | 36.56ms - 37.13ms | - | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -0.42ms - +0.52ms |
| branch | 776 kB | 36.42ms - 37.17ms | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -0.52ms - +0.42ms | - |


    
### element-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 756 kB | 349.76ms - 354.03ms | - | <strong>unsure 🔍</strong> <br> -1% - +0% <br> -4.70ms - +1.33ms |
| branch | 748 kB | 351.45ms - 355.71ms | <strong>unsure 🔍</strong> <br> -0% - +1% <br> -1.33ms - +4.70ms | - |


    
### lazy-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 757 kB | 53.19ms - 54.34ms | - | <strong>unsure 🔍</strong> <br> -3% - +0% <br> -1.41ms - +0.27ms |
| branch | 749 kB | 53.73ms - 54.95ms | <strong>unsure 🔍</strong> <br> -1% - +3% <br> -0.27ms - +1.41ms | - |


</details>


<details>
    <summary><strong>Firefox</strong></summary>

<a id="overlay"></a>

## overlay [_permalink_](#user-content-overlay)


    
### basic-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 763 kB | 713.89ms - 728.23ms | - | <strong>slower ❌</strong> <br> 2% - 5% <br> 14.99ms - 31.61ms |
| branch | 755 kB | 693.56ms - 701.96ms | <strong>faster ✔</strong> <br> 2% - 4% <br> 14.99ms - 31.61ms | - |


    
### directive-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 765 kB | 72.23ms - 73.13ms | - | <strong>unsure 🔍</strong> <br> -2% - +0% <br> -1.55ms - +0.15ms |
| branch | 757 kB | 72.66ms - 74.10ms | <strong>unsure 🔍</strong> <br> -0% - +2% <br> -0.15ms - +1.55ms | - |


    
### element-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 756 kB | 634.65ms - 643.11ms | - | <strong>slower ❌</strong> <br> 6% - 8% <br> 36.12ms - 46.28ms |
| branch | 748 kB | 594.88ms - 600.48ms | <strong>faster ✔</strong> <br> 6% - 7% <br> 36.12ms - 46.28ms | - |


    
### lazy-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 757 kB | 118.60ms - 124.96ms | - | <strong>slower ❌</strong> <br> 8% - 15% <br> 9.29ms - 15.95ms |
| branch | 749 kB | 108.17ms - 110.15ms | <strong>faster ✔</strong> <br> 8% - 13% <br> 9.29ms - 15.95ms | - |


</details>

File: tasks/build-tachometer-comment.js [hunk 3]
```diff
@@ -101,29 +101,46 @@ function formatDifference({ absolute, percentChange: relative }) {
...
+        const remote = result;
+        const remoteDifferences = formatDifference(remote.differences[i + 1]);
+        const remoteDifferencesString = `${remoteDifferences.label} <br> ${remoteDifferences.relative} <br> ${remoteDifferences.absolute}`;
+
+        const branch = results[i + 1];
+        const branchDifferences = formatDifference(branch.differences[i]);
+        const branchDifferencesString = `${branchDifferences.label} <br> ${branchDifferences.relative} <br> ${branchDifferences.absolute}`;
+
+        table.push(`
+### ${testName}
```
Comment (Westbrook) [inline tasks/build-tachometer-comment.js:125]: Could we add test level permalinks (but only if there is more than one set of results?

**Comment by najikahalsema** [comment]
@Westbrook How's this look? 

## Tachometer results


<details>
    <summary><strong>Chrome</strong></summary>

<a id="overlay"></a>

## overlay [_permalink_](#user-content-overlay)


    
### basic-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 694 kB | 483.04ms - 490.25ms | - | <strong>unsure 🔍</strong> <br> -1% - +2% <br> -2.47ms - +8.00ms |
| branch | 675 kB | 480.08ms - 487.68ms | <strong>unsure 🔍</strong> <br> -2% - +1% <br> -8.00ms - +2.47ms | - |


    
### directive-test [_permalink_](#user-content-directive-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 765 kB | 37.38ms - 38.17ms | - | <strong>unsure 🔍</strong> <br> -2% - +1% <br> -0.87ms - +0.21ms |
| branch | 768 kB | 37.74ms - 38.47ms | <strong>unsure 🔍</strong> <br> -1% - +2% <br> -0.21ms - +0.87ms | - |


    
### element-test [_permalink_](#user-content-element-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 756 kB | 359.83ms - 364.88ms | - | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -3.28ms - +3.52ms |
| branch | 748 kB | 359.96ms - 364.51ms | <strong>unsure 🔍</strong> <br> -1% - +1% <br> -3.52ms - +3.28ms | - |


    
### lazy-test [_permalink_](#user-content-lazy-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 757 kB | 54.69ms - 56.17ms | - | <strong>unsure 🔍</strong> <br> -2% - +2% <br> -0.84ms - +1.03ms |
| branch | 749 kB | 54.76ms - 55.92ms | <strong>unsure 🔍</strong> <br> -2% - +2% <br> -1.03ms - +0.84ms | - |


</details>


<details>
    <summary><strong>Firefox</strong></summary>

<a id="overlay"></a>

## overlay [_permalink_](#user-content-overlay)


    
### basic-test
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 763 kB | 751.29ms - 779.99ms | - | <strong>slower ❌</strong> <br> 2% - 6% <br> 13.48ms - 44.48ms |
| branch | 755 kB | 730.80ms - 742.52ms | <strong>faster ✔</strong> <br> 2% - 6% <br> 13.48ms - 44.48ms | - |


    
### directive-test [_permalink_](#user-content-directive-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 765 kB | 75.65ms - 77.47ms | - | <strong>slower ❌</strong> <br> 1% - 3% <br> 0.48ms - 2.56ms |
| branch | 757 kB | 74.54ms - 75.54ms | <strong>faster ✔</strong> <br> 1% - 3% <br> 0.48ms - 2.56ms | - |


    
### element-test [_permalink_](#user-content-element-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 756 kB | 664.37ms - 676.55ms | - | <strong>slower ❌</strong> <br> 5% - 8% <br> 34.31ms - 49.77ms |
| branch | 748 kB | 623.66ms - 633.18ms | <strong>faster ✔</strong> <br> 5% - 7% <br> 34.31ms - 49.77ms | - |


    
### lazy-test [_permalink_](#user-content-lazy-test)
| Version | Bytes | Avg Time | vs remote | vs branch |
|---|---|---|---|---|
| npm latest | 757 kB | 119.30ms - 125.54ms | - | <strong>slower ❌</strong> <br> 4% - 10% <br> 5.02ms - 11.58ms |
| branch | 749 kB | 113.10ms - 115.14ms | <strong>faster ✔</strong> <br> 4% - 9% <br> 5.02ms - 11.58ms | - |


</details>

File: tasks/build-tachometer-comment.js [hunk 4]
```diff
@@ -101,29 +101,50 @@ function formatDifference({ absolute, percentChange: relative }) {
...
+
+        const branch = results[i + 1];
+        const branchDifferences = formatDifference(branch.differences[i]);
+        const branchDifferencesString = `${branchDifferences.label} <br> ${branchDifferences.relative} <br> ${branchDifferences.absolute}`;
+
+        table.push(`
+### ${
+            testName.includes(`basic`)
+                ? `${testName}`
+                : `${testName} [_permalink_](#user-content-${testName})`
```
Comment (Westbrook) [inline tasks/build-tachometer-comment.js:128]: We should included the `packageName` in this link to reduce the chance of `testName` collisions between multiple packages.
```suggestion
                : `${testName} [_permalink_](#user-content-${packageName}-${testName})`
```

**Review by Westbrook — COMMENTED** [review]
Looks good after this last change!

**Review by Westbrook — APPROVED** [review]
:shipit: Woot!

**Pull request merged**